### PR TITLE
Migrate to `humantime` (phase 1)

### PR DESCRIPTION
**Pull request by squadgazzz**
# Description
Closes phase 1 from the https://github.com/cowprotocol/services/issues/2085. Since DEX solvers are about to migrate to a separate repo, it's a good time to change the config format. It will be much more difficult later.

# Changes
Support both plain and `humantime` formats to migrate our Pulumi config gradually.

## How to test
e2e + staging

### CODE REVIEW

File: crates/autopilot/src/arguments.rs [hunk 1]
```diff
@@ -376,6 +376,8 @@ impl FromStr for FeePolicyKind {
 }
 
 fn duration_from_days(s: &str) -> Result<Duration, ParseFloatError> {
-    let days = s.parse::<f64>()?;
-    Ok(Duration::from_secs_f64(days * 86_400.0))
+    humantime::parse_duration(s).or_else(|_| {
+        s.parse::<f64>()
+            .map(|days| Duration::from_secs_f64(days * 86_400.0))
+    })
```
Comment (fleupold) [inline crates/autopilot/src/arguments.rs:382]: Can we not remove this method in favor of `humantime::parse_duration` altogether? I think it's weird to have a method `duration_from_day` that you can call with "10m"

Or is this phase2 after the pulumi config update?

**Review by fleupold — APPROVED** [review]
Looks good, just a small confusion around the continuation of "from_days/from_seconds"

**Review by MartinquaXD — COMMENTED** [review]
I think I would not switch over in multiple phases.
The infra PR that changes from the old config to the new one can be prepared before hand and verified locally. And when this PR gets merged it can quickly be applied.
When we apply the changes together in prod next week we'll have no trouble.
So I think the only upside to splitting this change into multiple phases is avoiding a tiny downtime on staging which would be totally okay. It's a testing environment which is allowed to break from time to time.

**Comment by MartinquaXD** [comment]
> Why there won't be downtime as we would have in staging? infra and services are deployed at the same time in prod?

Exactly.

> Not sure how to test it locally before merging this PR.

You could probably log the content of the files pulumi would create if you were to apply the change and try to run your local system with those to see if things break. But that might be a bit involved if you want to make sure **everything** works.

Alternatively it should be possible to build a docker container from the PR branch and use that when preparing your infra change. Then you can play around with both changes together on `sepolia` for example until they work together.

IIRC in order to make CI build a container from your branch it should have a tag starting with `v` (see [CI](https://github.com/cowprotocol/services/blob/main/.github/workflows/deploy.yaml#L5)).

**Pull request closed without merging**